packages/eas-cli/src/worker/assets.ts

```diff
@@ -0,0 +1,170 @@
+/* eslint-disable async-protect/async-suffix */
+
+import { Gzip, GzipOptions } from 'minizlib';
+import { HashOptions, createHash, randomBytes } from 'node:crypto';
+import fs, { createWriteStream } from 'node:fs';
+import os from 'node:os';
+import path from 'node:path';
+import { pipeline } from 'node:stream/promises';
+import { pack } from 'tar-stream';
+
+/** Returns whether a file or folder is ignored */
+function isIgnoredName(name: string): boolean {
+  switch (name) {
+    // macOS system files
+    case '.DS_Store':
+    case '.AppleDouble':
+    case '.Trashes':
+    case '__MACOSX':
+    case '.LSOverride':
+      return true;
+    default:
+      // Backup file name convention
+      return name.endsWith('~');
+  }
+}
+
+/** Creates a temporary file write path */
+async function createTempWritePath(): Promise<string> {
+  const basename = path.basename(__filename, path.extname(__filename));
+  const tmpdir = await fs.promises.realpath(os.tmpdir());
+  const random = randomBytes(4).toString('hex');
+  return path.resolve(tmpdir, `tmp-${basename}-${process.pid}-${random}`);
+}
+
+/** Normalizes given sourcemap sources to relative paths for a given root path */
+function formatSourcemap(rootPath: string, data: Buffer | string): Buffer | string {
+  try {
+    const cwd = process.cwd();
```

Comment:
`process.cwd` might be a bit hard to test, it works fine but might be nice to have a clear "rootPath" vs something like "projectRoot" or "workspaceRoot" that defaults to `process.cwd`, but can be overridden?

Reply:
I think we should be using the `projectPath` here. Nice catch!

Reply:
Just checked, this is dead code and not needed anymore in the latest version 🎉 Forgot about that